osp/core/session/sparql_backend.py

```diff
@@ -69,6 +70,11 @@ def __enter__(self):
         """Enter the with statement."""
         return self
 
+    def __call__(self, **kwargs):
+        """Add kwargs to datatypes when class is called."""
+        self.datatypes.update(kwargs)
```

Comment:
Replace by `self.datatypes = kwargs` as we discussed, as otherwise every call would be remembered.